### PR TITLE
Личный проект: пока все дома

### PR DESCRIPTION
**Pull request by EDGARRUS**
---
:mortar_board: [Личный проект: пока все дома](https://up.htmlacademy.ru/javascript/18/user/529035/tasks/5)

:boom: https://htmlacademy-javascript.github.io/529035-keksobooking-18/2/

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+function arrayRandElement(arr) {
+  var rand = Math.floor(Math.random() * arr.length);
+  return arr[rand];
+}
+
+var generateData = function (counter) {
```
Comment (uladzimir-miadzinski) [inline js/main.js]: everything in javascript is Data/Object. you should use suitable names for variables and functions and don't use common words like objects/data/var/variable and etc.

File: index.html [hunk 2]
```diff
@@ -247,6 +247,6 @@ <h4 class="popup__type">Квартира</h4>
       <button class="error__button" href="#">Попробовать снова</button>
     </div>
   </template>
-
+<script src="js/main.js" type="text/javascript"></script>
```
Comment (uladzimir-miadzinski) [inline index.html]: you can omit  _type="text/javascript"_ cuz it is by default for script tag

File: js/main.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  var title = ['Luxury Apart', 'Good flat', 'My dear home', 'Great place', 'Wonderful house', 'Gold palace', 'Very cool home', 'Cheap flat in center'];
+  var description = ['Прекрасное место для отдыха', 'Вы полюбите это место как свой дом', 'Очень дешево и сердито', 'Вы не заходите уезжать из моего дома', 'Приезжай сейчас и я дам скидку 10%!', 'Хочешь красиво жить? Бронируй мое жилье', 'Красивый вид из окон на море', 'Квартира в самом центре города!'];
+  var checkin = ['12:00', '13:00', '14:00'];
+  var checkout = ['12:00', '13:00', '14:00'];
+  var features = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+  var houseInfo = [];
+
+  for (var i = 1; i <= counter; i++) {
+
+    var pinHeight = 65;
```
Comment (uladzimir-miadzinski) [inline js/main.js]: you should not define constant variables inside loops

File: js/main.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    var locationX = Math.floor(Math.random() * 1200) - pinHeight;
+    var locationY = Math.floor(Math.random() * 500) + 130 - pinHeight;
+
+    houseInfo.push({
+      'author': {
```
Comment (uladzimir-miadzinski) [inline js/main.js]: quotes ' is not necessary here because object properties are single one words

File: js/main.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+  return houseInfo;
+};
+
+var allHouses = generateHouses(8);
+
+var pinsList = document.querySelector('.map');
+pinsList.classList.remove('map--faded');
+
+var pinTemplate = document.getElementById('pin')
```
Comment (uladzimir-miadzinski) [inline js/main.js]: if you try to get nested element, you can use querySelector with selector like `#pin > .map__pin`

Comment (EDGARRUS) [reply js/main.js]: Там темплейт, так нельзя сделать же

**Pull request merged**